### ISSUE TITLE
Parallel execution of tasks.

### ISSUE THREAD

**Issue by takagi** [takagi/lake#5]
Support execute tasks in parallel to spawn multiple sub processes. It is supposed that underlying dependency among tasks would not be manageable for lake itself.

GNU make has `-j` option that specifies the number of tasks to run simultaneously.

```
$ man make
...
-j [jobs], --jobs[=jobs]
        Specifies the number of jobs (commands) to run simultaneously.  If
        there  is  more than one -j option, the last one is effective.  If
        the -j option is given without an argument, make  will  not  limit
        the number of jobs that can run simultaneously.
...
```

Now I implement it without any thread pooling.

**Comment by takagi** [comment]
For parallel task execution in `lake` command, local let to `*task*` dynamic variable curiously leads task not found error. To avoid this, I implement the following branches to choose issue/5.3.
- issue/5 Use LET% macro. [ok]
- issue/5.1 Bind *TASKS* to TASKS before jumping into threads.[ng]
- issue/5.2 Pass tasks explicitly as an argument. [ok]
- issue/5.3 Bind *TASKS* to *TASKS-IN-PROGRESS* during task execution. [ok]
- issue/5.4 Dispatch using execution context [ok]

**Comment by takagi** [comment]
How conditions that occur in child threads should be handled?

**Comment by guicho271828** [comment]
Are you writing it by yourself? Lparallel is a mature library that is capable of doing this kind of parallel dependency management. Lparallel's kernels specify the max number of threads, `task-handler-bind` captures the signals in child threads, `ptree` resolves the dependencies and compute the results in parallel. Sequential execution can be simulated with the number of threads = 1, which will greatly simplify the implementation (i.e. no need to change the code between parallel or sequential mode). If you consider refactoring, [`ptree` section of the author's manual](http://lparallel.org/ptrees/) would be useful.

**Comment by takagi** [comment]
OMG

**Comment by takagi** [comment]
Close by #50 .

**Comment by takagi** [comment]
@guicho271828 Ptree based simultaneous task execution, branch `issue/5.6`, will be merged into master soon. Thanks for your advice.
